src/app/utilities/ui-helper.ts

```diff
@@ -24,4 +26,23 @@ export class UIHelper {
       }
     }).bind(this), delay);
   }
+
+  /**
+   * The PrimeNg library does not handle selecting of a parent node
```

Comment:
Code documentation is most valuable if it at least describes what it does. Why it is implemented is also useful to note, but maybe as a second remark.
```javascript
/**
 * For a tree node, this function selects its ancestors (parent nodes) recursively. (General description, one-liner if possible)
 * 
 * Then a detailed description may follow, e.g., when this should be used, and when not. Why is this needed and is no standard library solution sufficient, preconditions on state or input, side effects, etc.
 * @param
 * @return
 * @throw
 */
```